lib/persistedQueryDefaults.js

```diff
@@ -0,0 +1,32 @@
+const crypto = require('crypto')
+
+const PersistedQueryDefaults = {
+  Prepared: (persistedQueries) => ({
+    isPersistedQuery: (r) => r.persisted,
+    getHash: (r) => r.query,
+    getQueryFromHash: async (hash) => persistedQueries[hash]
+  }),
+  PreparedOnly: (persistedQueries) => ({
+    isPersistedQuery: (r) => true,
+    getHash: (r) => r.persisted ? r.query : false, // Only support persisted queries
+    getQueryFromHash: async (hash) => persistedQueries[hash]
+  }),
+  Automatic: () => {
+    // Initialize only in the scope of this server instance
+    const AUTOMATIC_PERSISTED_QUERIES = {}
```

Comment:
This must be a very large LRU cache instead, otherwise we are exposing ourselves to a DoS attack by memory exhaustion. check out https://www.npmjs.com/package/mnemonist.

Reply:
This is a good point, I'll look into implementing it.

Reply:
Have added